Wrappers/Python/ccpi/framework/framework.py

```diff
@@ -113,14 +134,34 @@ def __str__ (self):
         repres += "voxel_size : x{0},y{1},z{2}\n".format(self.voxel_size_x, self.voxel_size_y, self.voxel_size_z)
         repres += "center : x{0},y{1},z{2}\n".format(self.center_x, self.center_y, self.center_z)
         return repres
+    
     def allocate(self, value=0, dimension_labels=None):
         '''allocates an ImageData according to the size expressed in the instance'''
-        if dimension_labels is None:
-            dimension_labels = self.dimension_labels
-        out = ImageData(geometry=self, dimension_labels=dimension_labels)
-        if value != 0:
-            out += value
+        out = ImageData(geometry=self)
+        if isinstance(value, Number):
+            if value != 0:
+                out += value
+        else:
+            if value == 'random':
+                out.fill(numpy.random.random_sample(self.shape))
+            elif value == 'random_int':
+                out.fill(numpy.random.randint(100, size=self.shape))
+        if dimension_labels is not None:
+            if dimension_labels != self.dimension_labels:
+                return out.subset(dimensions=dimension_labels)
         return out
+    
+    def __len__(self):
+        return self.length
+    
+    def shape(self):
```

Comment:
is this method needed? We have access to the shape with the member `shape` anyway.

Reply:
Actually, both methods `shape` and `__len__` just return two members of the class. Do we really need them as methods?

What's wrong with 

```python
ig.length
# wrt 
len(ig)

ig.shape
# wrt
ig.shape()
```